frontend/app/work_packages/controllers/work-package-details-controller.js

```diff
@@ -172,13 +174,53 @@ module.exports = function($scope,
     return !!($scope.workPackage && $scope.workPackage.embedded.watchers !== undefined);
   };
 
-  function displayedActivities(workPackage) {
+  $scope.matchesActivityType = function(activity, type) {
```

Comment:
while this is a nice idea, it's not readable very well.

I coul image giving the underlying controller some information about the types and then define some more readable helper functions for the viewModel, i.e. `isRevisionType`